src/MPS.jl

```diff
@@ -541,6 +542,115 @@ function mixed_canonize!(tn::AbstractMPO, orthog_center)
     return tn
 end
 
+"""
+    evolve!(ψ::AbstractAnsatz, mpo::AbstractMPO; threshold=nothing, maxdim=nothing, normalize=true)
+
+Evolve the [`AbstractAnsatz`](@ref) `ψ` with the [`AbstractMPO`](@ref) `mpo` along the output indices of `ψ`.
+If `threshold` or `maxdim` are not `nothing`, the tensors are truncated after each sweep at the proper value.
+"""
+function evolve!(ψ::AbstractAnsatz, mpo::AbstractMPO; threshold=nothing, maxdim=nothing, normalize=true)
+    evolve!(form(ψ), ψ, mpo; threshold, maxdim, normalize)
```

Comment:
@starsfordummies Tomorrow I will add here the `replace!` so this `evolve!` function does not change the output indices, after that, we can merge!